### ISSUE TITLE
Cordova build issues and broken resulting APK.

### ISSUE THREAD

**Issue by CompletelyGeneric** [MaKleSoft/padlock#95]
Versions:
cordova: 7.0.1
npm: 5.3.0
node: 8.2.1
Running the APK on an OP3T running the latest LOS.
Building on windows 10

cordova complains about the `cordova` directory not being a cordova project unless you `mkdir www plugins platforms` first  

```
$ cordova platform add android  
Using cordova-fetch for cordova-android@^6.2.3  
Adding cordova-plugin-wkwebview-engine project...  
Error: Your cordova-plugin-wkwebview-engine platform does not have Api.js  
```

Interestingly enough, running `cordova platform add android` exactly three times results in success, albeit with a few warnings.  

```Android project created with cordova-android@6.2.3
Discovered plugin "com.verso.cordova.clipboard" in config.xml. Adding it to the project
Failed to restore plugin "com.verso.cordova.clipboard" from config.xml. You might need to try adding it again. Error: Failed to fetch plugin com.verso          .cordova.clipboard@https://github.com/VersoSolutions/CordovaClipboard.git via registry.
Probably this is either a connection problem, or plugin spec is incorrect.
Check your connection and plugin name/version/URL.
Error: cmd: Command failed with exit code 1 Error output:
npm ERR! cb() never called!
```

Additionally, `com.wongatech.cordova-disable-nsurl-cache` errors out in the same way. 

Executing `cordova build android --device --xwalk64bit` completes the build successfully, while throwing the same errors as above, however, [the resulting APK is quite interesting](https://my.mixtape.moe/cxwbpl.mp4)  
Note, that the application still correctly blocks screenshots.

**Comment by MaKleSoft** [comment]
Thanks for the detailed report! What's the OS of your build environment?

**Comment by CompletelyGeneric** [comment]
Windows 10 x64

I've never written node before, but It seems to me that the error is coming from polymer-bundler not liking to parse windows absolute paths. If I invoke it with relative paths from the root folder like so `node_modules/polymer-bundler/lib/bin/polymer-bundler.js --inline-scripts --inline-css app/index.html > cordova/www/index.html` , it seems to function as intended.

**Comment by MaKleSoft** [comment]
Good catch. You're running the build command from the cordova directory,
right?

**Comment by CompletelyGeneric** [comment]
Yep.

**Comment by MaKleSoft** [comment]
Ok, so it seems `polymer-bundler` is having trouble accepting an absolute url for the source document (see https://github.com/Polymer/polymer-bundler/issues/521). I'll change the invocation to accommodate for that.

As for the other warnings, those are separate issues. The 'not a cordova project' error is due to the stupid requirement that the project needs to have a `www` directory, even if it is empty. I'll add a dummy file to source control to make sure that directory exists when the project is cloned. Also, it turns out that you have to explicitly run `cordova prepare` before running `cordova build` for the first time since otherwise it won't automatically add the platforms specified in the `config.xml`. I'll add that to the readme.

Thanks for pointing all of this out to me! This is the problem if you work with a single working copy. There is always some state sneaking in that makes it work locally but causes problems when trying to make things work with a fresh clone.

**Comment by CompletelyGeneric** [comment]
Awesome! Thanks a bunch.

**Comment by MaKleSoft** [comment]
No problem. Let me know if everything works for you now!

**Comment by CompletelyGeneric** [comment]
The same error still occurs. I've traced it down to weird behavior with how polymer-bundle is invoked via `exec()`.  The solution seems to be specifying that polymer-bundle.js is to be invoked with `node`.   

Oddly enough, this is only needed when invoking it via node's `exec()` or in `cmd.exe`.  
Invoking the call to `polymer-bundler.js` in powershell works perfectly.  

I've submitted a PR that fixes this.
